### PR TITLE
[MXNET-8230] test_operator_gpu.test_rms fails

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##
Fix flaky test#8230, test_rms fails.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Fix flaky test#8230, test_rms fails.

This flaky test was due to compare result using mx.test_utils.almost_equal() with high precision.
Reducing tolerance to 1e-1 for float16 and 1e-2 for float32. The test passed 10K iterations after this fix.

### CODE REVIEW

File: tests/python/unittest/test_optimizer.py [hunk 1]
```diff
@@ -835,8 +835,8 @@ def update(self, index, weight, grad, state):
         if self.clip_weights:
              mx.ndarray.clip(weight, -self.clip_weights, self.clip_weights, out=weight)
 
-@unittest.skip("Test fails intermittently. Temporarily disabled until fixed. Tracked at https://github.com/apache/incubator-mxnet/issues/8230")
-@with_seed(0)
+# @unittest.skip("Test fails intermittently. Temporarily disabled until fixed. Tracked at https://github.com/apache/incubator-mxnet/issues/8230")
```
Comment (haojin2) [inline tests/python/unittest/test_optimizer.py]: If you're getting rid of the line please delete it.

File: tests/python/unittest/test_optimizer.py [hunk 2]
```diff
@@ -848,6 +848,13 @@ def test_rms():
     wd_options = [{}, {'wd': 0.03}, {'wd': 0.05}, {'wd': 0.07}]
     mp_options = [{}, {'multi_precision': False}, {'multi_precision': True}]
     for dtype in [np.float16, np.float32]:
+        if (dtype == np.float16):
+            rtol=1e-1
+            atol=1e-1
+        else:
+            rtol=1e-2
+            atol=1e-2
```
Comment (haojin2) [inline tests/python/unittest/test_optimizer.py]: I would prefer:
```Python
rtol = 1e-1 if dtype is np.float16 else 1e-2
atol = 1e-1 if dtype is np.float16 else 1e-2
```

File: tests/python/unittest/test_optimizer.py (hunk 1 above)
Comment (lupesko) [reply tests/python/unittest/test_optimizer.py]: +1

File: tests/python/unittest/test_optimizer.py (hunk 2 above)
Comment (lupesko) [reply tests/python/unittest/test_optimizer.py]: IMO the way @frankfliu wrote it is clearer and easier to parse.
But would be great to have a comment explaining it (it was learned by trial and error, right?)

**Comment by lupesko** [comment]
@frankfliu thanks for contributing the fix! 
Can you please update the issue description:
- Tick all the boxes to show that you addressed the standard PR requirements (you can remove those that are not relevant)
- Under "Changes", explain the change briefly.

File: tests/python/unittest/test_optimizer.py (hunk 1 above)
Comment (frankfliu) [reply tests/python/unittest/test_optimizer.py]: done

File: tests/python/unittest/test_optimizer.py (hunk 2 above)
Comment (frankfliu) [reply tests/python/unittest/test_optimizer.py]: done

**Pull request merged**